File: tutorials/testing/python/color_grid.py
from bridges.color_grid import *
from bridges.bridges import *
from bridges.color import *
import sys

def main():
    # This example illustrates using the Bridges color grid
    # We will build a checker grid using two different colors


    # create the Bridges object, set credentials
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # title, description
    bridges.set_title("A CheckerBoard Example Using Grid Type")
    bridges.set_description("This is an example of the color grid with height and width 10. "
        +   "This example alternates between blue and red starting with red at the top left corner of the grid.")


    blue = Color(col_name="blue")
    red = Color(col_name="red")

    # create a 100x100 color grid
    width = 100
    height = 100
    cg = ColorGrid(width, height, red)

    # create a checkerboard pattern
    # 10x10 grid
    num_squares_x = 10
    num_squares_y = 10

    sq_width = width / num_squares_x
    sq_height = width / num_squares_y

    for j in range(num_squares_y):
        for k in range(num_squares_x):

            # use even/odd value of pixel to figure out the color of the squares
            x_even = k % 2 == 0
            y_even = j % 2 == 0

            if (y_even == True):
                if (x_even == True):
                    col = red
                else:
                    col = blue
            else:
                if (x_even == True):
                    col = blue
                else:
                    col = red

            # find the address of the square
            origin_x = k * sq_width
            origin_y = j * sq_height

            # color the square
            for y in range(int(origin_y), int(origin_y + sq_height)):
                for x in range(int(origin_x), int(origin_x + sq_width)):
                    cg.set(x, y, col)
                    x = x + 1
                y = y + 1

    bridges.set_data_structure(cg)

    bridges.visualize()


if __name__ == "__main__":
    main()

File: tutorials/testing/c++/color_grid.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "Color.h"
#include "ColorGrid.h"

using namespace bridges;

// This example illustrates using the Bridges color grid
// We will build a checker grid using two different colors

int main(int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// set title, description
	bridges.setTitle("A CheckerBoard Example Using Grid Type");
	bridges.setDescription("This is an example of the color grid with height and width 10. "
		"This example alternates between blue and red starting with red at the top left corner of the grid.");
	int width = 100, height = 100;

	ColorGrid cg(width, height, Color("red"));

	// create a checkerboard pattern
	// 10 x 10 grid
	const int num_squares_x = 10;
	const int num_squares_y = 10;

	// compute square dimensions
	const int sq_width = width / num_squares_x, sq_height = width / num_squares_y;

	for (int j = 0; j < num_squares_y;  j++)
		for (int k = 0; k < num_squares_x;  k++) {
			// use even/odd of pixel to figure out the color of the square
			bool x_even = (k % 2) == 0;
			bool y_even = (j % 2) == 0;

			string col;
			if (y_even)
				col = (x_even) ? "red" : "blue";
			else
				col = (x_even) ? "blue" : "red";

			// find the address of the square
			int origin_x = k * sq_width;
			int origin_y = j * sq_height;

			// color the square
			for (int row = origin_y; row < origin_y + sq_height; row++)
				for (int column = origin_x; column < origin_x + sq_width; column++)
					cg.set(row, column, Color(col));
		}

	bridges.setDataStructure(&cg);

	bridges.visualize();

	return 0;
}
